include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -155,6 +155,21 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         .wait();
 }
 
+template <typename Iterator1, typename Iterator2, std::enable_if_t<std::is_same_v<Iterator1, Iterator2>, int> = 0>
+constexpr bool
+__is_equal_iterators(Iterator1 it1, Iterator2 it2)
+{
+    return it1 == it2;
+}
+
+template <typename Iterator1, typename Iterator2, std::enable_if_t<!std::is_same_v<Iterator1, Iterator2>, int> = 0>
+constexpr bool
+__is_equal_iterators(Iterator1 it1, Iterator2 it2)
+{
+    static_assert("In-place exclusive scan works correctly only if an input and an output iterators are the same type.");
```

Comment:
And should not be any assert here - just a comment.
Otherwise it "kills" the regular implementation  - "else" code branch (191)

Reply:
FYI: in our tests, there are no cases when it's happened now in our tests.
If we will have a simple comment, depending on iterator types we have correct/incorrect working code without compile errors.
But up to you.

Reply:
Fixed

Reply:
> in our tests

"Unfortunately", a user runs not only our tests)...

Reply:
Yes, sure, and in this case better to have some info is our algorithm works correctly or not.
How are you going to give this use to oneDPL users? Simple by comment?

Reply:
> we have correct/incorrect working code without compile errors.

Unfortunately, there is no way to check it whether compile time or run time in a general case.

Reply:
I believe we may check that ```std::address()``` is applicable for current iterator type and in this case use it too before ```return false``` in ```__is_equal_iterators``` function.